packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -218,74 +204,91 @@ const BulkActionsBar = ({
     }
   };
 
-  const handleConfirmPublishAll = async () => {
+  const toggleUnpublishModal = () => {
+    if (dialogToOpen === 'unpublish') {
+      setDialogToOpen(null);
+    } else {
+      setDialogToOpen('unpublish');
+      trackUsage('willBulkUnpublishEntries');
+    }
+  };
+
+  const handleBulkDelete = async () => {
     try {
       setIsConfirmButtonLoading(true);
-      await onConfirmPublishAll(selectedEntries);
-      clearSelectedEntries();
-      setIsConfirmButtonLoading(false);
+      await onConfirmDeleteAll(selectedEntries);
     } catch (err) {
+      // The error is handled in content-manager/pages/ListView/index.js
+      // handleConfirmDeleteAllData catch block
+    } finally {
       setIsConfirmButtonLoading(false);
+      toggleDeleteModal();
+      clearSelectedEntries();
     }
-    handleToggleShowPublishAllModal();
   };
 
-  // Bulk unpublish
-  const handleToggleShowUnpublishAllModal = () => {
-    if (dialogToOpen === 'unpublish') {
-      setDialogToOpen(null);
-    } else {
-      setDialogToOpen('unpublish');
-      trackUsage('willBulkUnpublishEntries');
+  const handleBulkPublish = async () => {
+    try {
+      setIsConfirmButtonLoading(true);
+      await onConfirmPublishAll(selectedEntries);
+    } catch (error) {
+      // The error is handled in content-manager/pages/ListView/index.js
+      // bulkPublishMutation onError callback
+    } finally {
+      setIsConfirmButtonLoading(false);
+      togglePublishModal();
+      clearSelectedEntries();
```

Comment:
For some reason this has to go last otherwise I get this error:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
 ```

It's not ideal because I don't think we should clear the selected entries if there was an error, but I'm struggling to find a solution.

Reply:
🤔 weird. surely the `togglePublishModal` is what hides the component no? So `clearSelectedEntries` should in theory still work...

Reply:
Yeah, actually now noticing another strange behavior... if you comment toggleModal, clearSelectedEntries will still somehow close the modal...

Reply:
@markkaylor probably because of [this line](https://github.com/strapi/strapi/blob/b83c8d9320922e69862e0507d96e79cabe765a08/packages/core/helper-plugin/src/components/DynamicTable/index.js#L137)